### ISSUE TITLE
Lazy routes not recognized when using array import

### ISSUE THREAD

**Issue by csvn** [angular/angular#17531]
## I'm submitting a ...

<pre><code>
[ ] Regression (behavior that used to work and stopped working in a new release)
[x] Bug report 
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
Angular does not recognize lazy routes when they are imported like this:

```
@NgModule({
  exports: [RouterModule],
  imports: [
    [RouterModule.forRoot(routes)]
  ]
})
export AppRoutingModule {}
```
_(See more motivation below)_

## Expected behavior
Lazy loaded routes are recognized with the entire syntax that Angular supports (including imports with arrays).

## Minimal reproduction of the problem with instructions
Plunkr does not work well for this issue, but I created a small ng-cli repo here:
https://github.com/csvn/ng-lazy-route-issue

1. Clone and npm install
2. Start with `npm start`
3. Clicking on the 'lazy' link will display error

## What is the motivation / use case for changing the behavior?
The example above is contrived, but the use case I have is that I want to define routes in a library module that is imported with `forRoot()`. I tried to have the `forRoot()` method return an array of imports, but this broke the lazy routes. I wanted to do this because the library will wrap the routes in a common parent that has some auth related functionality and more.

There are many examples where an array of `imports/providers/exports/declarations/etc` are used, e.g.: https://github.com/angular/angular/blob/master/packages/forms/src/form_providers.ts#L23.

The `NgModule` class in Typescript also specify that `any[]` may be used for most fields: https://github.com/angular/angular/blob/master/packages/core/src/metadata/ng_module.ts#L132.

If an app is started with `imports: [RouterModule.forRoot(routes)]`, and then changed to `imports: [ [RouterModule.forRoot(routes)] ]` everything will still work fine, as the chunk is already created. So there doesn't seem to be any issues running with the latter syntax, just for recognizing lazy routes.

The reason chunks won't be created due to `__NGTOOLS_PRIVATE_API_2.listLazyRoutes()` (from `@angular/compiler-cli`) not finding any lazy routes. I am not sure if this is also the issue for when using `ng build --aot`, as it has the same issue as JIT.

## Please tell us about your environment

<pre><code>
Angular version: 4.2.2

Browser:
- [x] Chrome (desktop) version 59
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 8.1.0
- Platform: Windows

Others:

</code></pre>

**Comment by csvn** [comment]
Thought I could provide the more realistic usage scenario here. This is roughly what we are trying to do.

A shared library for creating a chromed/shell app (common header, login, auth, etc) should provide the following:
```typescript
@NgModule({
  exports: SHARED_STUFF,
  imports: OWN_STUFF
})
export class AppChromeLibraryModule {
  static forRoot(wrappedRoutes: Routes, config?: ExtraOptions): ModuleWithProviders[] {
    return [
      RouterModule.forRoot(chromeRoutes(wrappedRoutes), config),
      { ngModule: AppChromeLibraryModule, providers: CHROME_PROVIDERS }
    ];
  }
}
```

This can then be used to create an app like this:

```typescript
const routes: Routes = {
  // routes for logged in users ...
};

@NgModule({
  imports: [
    AppChromeLibraryModule.forRoot(routes)
  ]
})
export class AppRoutingModule {}
```

**Comment by Toxicable** [comment]
I've tried to reproduce it in a plunkr bu so far been unsuccessful. Could you verify what I've done matches your issue? This is to isolate the problem from the CLI
http://plnkr.co/edit/mU97Xt4HqQfu96A4T3PN?p=preview

**Comment by csvn** [comment]
@Toxicable  I don't believe it can be reproduced in a Plunkr, since it does not use the same pipeline as ng-cli (SystemJS in browser vs Webpack with dev-server).

https://github.com/angular/angular-cli/blob/master/packages/%40ngtools/webpack/src/plugin.ts#L495-L500 is the code that is returning an empty object instead of an object with lazy-path-mappings, and that's being used as part of a webpack plugin. The offending Angular package for this is `@angular/compiler-cli`.

**Comment by csvn** [comment]
TL;DR: The code that detects lazy loaded routes in Angular, for creating Webpack chunks, seem to fail when routing is placed inside an array in `imports`.

**Comment by hansl** [comment]
This is an issue with the lazy route detection part of the Compiler, that is used in the CLI / webpack plugin.

It can only be fixed in this repo, so I'm redirecting this to the proper team and will hopefully fix soon (or update this issue).
